tsconfig.json

```diff
@@ -2,8 +2,28 @@
   "extends": ["@tsconfig/node18/tsconfig"],
   "compilerOptions": {
     "declaration": true,
+    "declarationMap": true,
```

Comment:
just for debugging purpose, since we are making a library it is always nice to use it for things like "Go to Definition." and we can always exclude those files from build.

Anyway is not a big deal and in our case we don't have a lot of files so I can remove it if you like